lms/views/api/moodle/files.py

```diff
@@ -29,12 +35,61 @@ def list_files(_context, request):
     permission=Permissions.API,
 )
 def via_url(_context, request):
-    token = request.find_service(MoodleAPIClient).token
+    course_copy_plugin = request.product.plugin.course_copy
```

Comment:
The following code handles a few cases separately is very similar between pages and files. Also very similar to D2L.